code/controllers/subsystem/garbage.dm

```diff
@@ -284,15 +284,21 @@ SUBSYSTEM_DEF(garbage)
 ///
 /// Datums passed to this will be given a chance to clean up references to allow the GC to collect them.
 /proc/qdel(datum/D, force=FALSE, ...)
+	if(isweakref(D))
+		var/datum/weakref/weakref = D
+		D = weakref.resolve()
+		if(!D)
+			return
+
 	if(!istype(D))
```

Comment:
Should be below this

Reply:
We discussed it on discord and it shouldnt since you can have weakrefs to things that aren't datums @Cyberboss

Reply:
Yes, but isn't the weakref itself always a /datum?

Reply:
wait nevermind